weldx/transformations.py

```diff
@@ -1021,11 +1021,23 @@ def get_local_coordinate_system(
         path = nx.shortest_path(
             self.graph, coordinate_system_name, reference_system_name
         )
+        path_edges = []
+        for i in range(len(path) - 1):
+            path_edges.append((path[i], path[i + 1]))
+
+        time_union = self.time_union(path_edges)
+
         lcs = self.graph.edges[path[0], path[1]]["lcs"]
+        if time_union is not None:
+            lcs = lcs.interp_time(time_union)
+
         length_path = len(path) - 1
         if length_path > 1:
             for i in np.arange(1, length_path):
-                lcs = lcs + self.graph.edges[path[i], path[i + 1]]["lcs"]
+                lcs_rhs = self.graph.edges[path[i], path[i + 1]]["lcs"]
+                if time_union is not None:
+                    lcs_rhs = lcs_rhs.interp_time(time_union)
+                lcs = lcs + lcs_rhs
```

Comment:
We can do some list operations here 😃 
```python
        path_edges = [edge for edge in zip(path[:-1], path[1:])]

        time_union = self.time_union(path_edges)

        lcs = self.graph.edges[path[0], path[1]]["lcs"]
        if time_union is not None:
            lcs = lcs.interp_time(time_union)

        for edge in path_edges[1:]:
            lcs_rhs = self.graph.edges[edge[0], edge[1]]["lcs"]
            if time_union is not None:
                lcs_rhs = lcs_rhs.interp_time(time_union)
            lcs = lcs + lcs_rhs
```

Reply:
Added some further improvements. Now it looks really nice 😄